### DISCUSSION TITLE
504 timeout on server action

### DISCUSSION THREAD

**Discussion by graemen16** [vercel/community#5478] [Help]
### Summary

I'm working on 'CRUD heavy' app, currently next 14.0.2, app directory with server actions on Vercel Pro plan. The app includes a server action that makes a sequence of 15 to 20 CRUD calls on a database (Vercel Postgres). It works as expected running on localhost (in approx 7 seconds) but I'm getting 504 timeout after 15 seconds on deployed version. 
 The runtime log shows this as a Serverless Function, points to the route of the client page that called the server action and references execution duration / limit as 15.02s / 15s. Is there a way to set the max duration for server actions, i.e. functions not called via an api route? I could only find documentation on how to set max duration for api routes.

### Steps to Reproduce

 The runtime log shows this as a Serverless Function, points to the route of the client page that called the server action: '/settings/add-view-col' and references execution duration / limit as 15.02s / 15s.

**Comment by simonsayscode** [comment]
I found this comment [here ](https://www.reddit.com/r/nextjs/comments/1anvdpw/comment/kpv7o4o/?utm_source=reddit&utm_medium=web2x&context=3) from @leerob that states that Server Actions inherit the runtime of the page that they're called from. You can increase the `maxDuration` there and that should apply to the actions called.
